_posts/en/2019-09-19-haproxy-small-but-handy.md

```diff
@@ -0,0 +1,222 @@
+---
+layout: post
+title: HAProxy - small but handy
+published: true
+lang: en
+lang-ref:   haproxy-small-handy
+date:      2019-09-16 08:00:00 +0100
+author:    tlewandowski
+interpreter: Piotr Żurawski
+image:     /assets/img/posts/2019-09-19-haproxy-mala-rzecz-a-cieszy/haproxy.jpg
+tags:
+- programming
+- load-balancing
+- ha
+- server
+- network
+---
+
+
+HAProxy is open source software that is most often used as a reverse proxy to ensure load balancing and high availability of application servers.
+
+Instead of connecting directly to the servers of an application, clients (e.g. of a browser) do it from a reverse proxy, which uses additional rules to pass the requests to the application servers and send the responses back to the client.
+
+The project has been actively developed since 2002 and has seen increased popularity due to the proliferation of distributed systems, in particular architectures based on microservices. It is in use by such household names as Digital Ocean, GitHub, Dropbox, Instagram or StackOverflow. It is also a component of the OpenShift containerization platform.
+
+The software is available for all popular Linux distributions, also as a Docker image.
+
+## Typical usages of HAProxy
+
+### Ensuring load balancing
+One of the basic usages of HAProxy is as a software load balancer. 
+With several nodes in an application, the traffic between them needs to be managed. 
+To do that you need to declare an object called backend in the HAProxy configuration, which will represent a cluster of the application servers.
+Next, you need to declare an object called frontend, to which clients will be directed and frontend-to-backend traffic rules:
+```
+frontend my-load-balancer
+    bind 172.19.0.1:81
+    default_backend my-application-servers
+    
+backend my-application-servers
+    balance roundrobin
+    server my-app-server-1 172.19.0.2:8080
+    server my-app-server-2 172.19.0.3:8080
+    server my-app-server-3 172.19.0.4:8080   
+```
+As a result of the above configuration, the connection on endpoint 172.19.0.1:81 will be redirected to one of the servers on a round-robin basis.
+
+###Ensuring High Availability
```

Comment:
brakuje odstępu i kiepsko się to renderuje

![image](https://user-images.githubusercontent.com/1863167/121847787-680c1280-cce9-11eb-9864-7fcc9014bd1f.png)

Reply:
Dzięki, przegapiłem